core/src/io/anuke/mindustry/world/BaseBlock.java

```diff
@@ -40,6 +44,13 @@ public boolean shouldConsume(Tile tile){
         return true;
     }
 
+    public float getPowerProduction(Tile tile){
+        if(tile.block().hasPower && tile.block().outputsPower){
```

Comment:
Why... are you doing all these checks specifically for power generators when the point is for `PowerGenerator` to override this method?

Reply:
ok turns out I'm dumb